src/pudl/transform/ferc1.py

```diff
@@ -3866,3 +3885,474 @@ def plants_steam_ferc1(
         transformed_fuel=fuel_ferc1,
     )
     return convert_cols_dtypes(df, data_source="ferc1")
+
+
+##################################
+# Calculations/Metadata Management
+##################################
+
```

Comment:
everything below here is my very WIP code that will hopefully set us up to make these exploded tables. I don't know where this should actually end up living? It feels more like `output.ferc1` stuff. It is very dependent on the transform params and I wasn't sure originally if some of this translation should happen during the transforms themselves. The more I muck with it, the more it is clear that the calculations need to care about more than just one table so I feel less drawn to that idea than I did originally.